commit #2

fe327fe opened the pull request.

fe327fe committed:
commit #2

pavel-ulan commented on src/main/java/PageObject/RegisterPage.java:
```diff
@@ -0,0 +1,102 @@
+
+    // локатор поля для ввода имени
+    @FindBy(how = How.XPATH,using =("//label[@class='input__placeholder text noselect text_type_main-default'][text()" +
+            "='Имя']/parent::div/input"))
```
я бы сказал, что локаторы "на тоненького". Сложный, с переходами к родителю, да еще и `noselect` фигурирует (что при определенных условиях может сломать тест.
По хорошему, стоит эти 3 локатора переделать.

pavel-ulan commented on src/main/java/TestModel/User/UserCredentials.java:
```diff
@@ -0,0 +1,56 @@
+@Builder
+@Setter
+@Data
+@AllArgsConstructor
```
Похвально, что ты используешь lombok, но например в этом классе у тебя явно прописан конструктор через user'а. И возникает вопрос, зачем тебе аннотация AllArgs (если ты нигде не используешь этот конструктор)? И следом помни, что если у тебя нет аннотации для конструкторов (или конструктора. кстати, я не уверен, твой конструктор будет учитываться ломбуком или нет 🤔 ), но есть аннотация Data, то конструктор без параметров будет создан автоматом.
Я это все пишу скорее не как замечание, а как пунктик на подумать. Маловероятно, что тебе когда-то понадобиться так тонко настраивать точки доступа к модели, но держи в голове, что ты создаешь, зачем и не дублирует (или замещает) ли одно другое? Просто из-за этого могут быть неожиданные эффекты.

pavel-ulan commented on src/test/java/TestRegisterNewUser.java:
```diff
@@ -0,0 +1,78 @@
+                .extract()
+                .path("accessToken");
+
+        userClient.delete(user.getEmail(), bearerToken);
```
есть косяк. если тест сломается, то до этой строчки не дойдет и пользователь не будет удален.
такие действия надо выносить в постстепы (ну или на крайняк заверни все в softAsserts. тогда, даже если ожидание сломается, то до строчки с удалением пользователя дойдет).

pavel-ulan commented on src/test/java/TestUserLogin.java:
```diff
@@ -0,0 +1,113 @@
+    public void teardown(){
+
+        userClient.delete(user.getEmail(), bearerToken);
+    }
```
а тут правильный подход

pavel-ulan commented on src/test/java/TestBurgerIngredients.java:
```diff
@@ -0,0 +1,69 @@
+        // клик по ссылке <Начинки>
+        // клик по ссылке <Соусы>
+        // клик по ссылке <Булки>
+        // проверяем наличие текстового элемента <Булки>
```
Все клево, но задание звучит иначе. Надо проверить не текст, а 
`Проверь, что работают переходы к разделам:
- «Булки»,
- «Соусы»,
- «Начинки».`
что, несколько иное...
Попробуй проскролить разделы и смотреть как меняются имена классов в верстке у разделов. И построй проверку "в каком разделе я нахожусь" на этой основе.

pavel-ulan reviewed (changes requested):
Вот тут немного поплыл. Посмотри замечания, поправь и присылай на повторную проверку.
(но если в целом, то работа хороша)

fe327fe committed:
commit #3

и снова здравствуйте!
вроде бы все, поправил, что ты просил
не уверен, конечно, что наиболее оптимальным способом...

please, review :)
